docs/elasticsearch.txt

```diff
@@ -84,6 +84,126 @@ JanusGraph only uses default values for `index-name` and `health-request-timeout
 
 The REST client accepts the `index.[X].bulk-refresh` option. This option controls when changes are made visible to search. See https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html[?refresh documentation] for more information.
 
+==== REST Client HTTPS Configuration
+
+SSL support for HTTP can be enabled by setting the `index.[X].elasticsearch.ssl.enabled` configuration option to `true`. Note that depending on your configuration you may need to change the value of `index.[X].port` if your HTTPS port number is different from the default one for the REST API (9200).
+
+When SSL is enabled you may also configure the location and password of the truststore. This can be done as follows:
+
+[source, properties]
+----
+index.search.elasticsearch.ssl.truststore.location=/path/to/your/truststore.jks
+index.search.elasticsearch.ssl.truststore.password=truststorepwd
+----
+
+Note that these settings apply only to Elasticsearch REST client and do not affect any other SSL connections in JanusGraph.
+
+Configuration of the client keystore is also supported:
+
+[source, properties]
+----
+index.search.elasticsearch.ssl.keystore.location=/path/to/your/keystore.jks
+index.search.elasticsearch.ssl.keystore.storepassword=keystorepwd
+index.search.elasticsearch.ssl.keystore.keypassword=keypwd
+----
+
+Any of the passwords can be empty.
+
+
+==== REST Client Basic HTTP Authentication
+
+Basic HTTP Authentication is available regardless of the state of SSL support.  Authentication realm value is optional. If it is not provided, the credentials will match any authentication realm value.
+
+
+[source, properties]
+----
+index.search.elasticsearch.http.auth.type=basic
+index.search.elasticsearch.http.auth.basic.username=httpuser
+index.search.elasticsearch.http.auth.basic.password=httppassword
+index.search.elasticsearch.http.auth.basic.realm=myesrealm
```

Comment:
Thanks a lot for additional testing! I was thinking about the hostname verification too. I am not a big fan of this (kind of defeats the purpose of SSL) and I prefer to test with my own CA and the certificates issued by this CA - since I have the configurable trust store :) But it is a very simple change, adding it now.